README.md

```diff
@@ -123,152 +125,244 @@ Congratulations! You can now send remote push notifications to your device from
+	// perform custom processing
+	
+	// ...
+	
+	return InAppShowResponseShow; // or InAppShowResponseSkip
+}
+
+// ...
+	
+// Now set this custom class in IterableConfig
+IterableConfig *config = [[IterableConfig alloc] init];
+config.inAppDelegate = self; // or other class implementing the protocol
+[IterableAPI initializeWithApiKey:@"YOUR API KEY" launchOptions:launchOptions config:config];
+```
+
+###### Getting the local queue of in-app messages
+Until they are consumed by the app, all in-app messages that arrive from the server are stored in a local queue. To access that local queue, use the read-only `IterableAPI.inAppManager` property (which conforms to the `InAppManager` protocol). By default, all in-app messages in the local queue will be consumed and removed from this queue. To keep in-app messages around after they are shown, override the default behavior (as described above).
+	
+**Swift**
+	
+```swift
+// Get the in-app messages list
+let messages = IterableAPI.inAppManager.getMessages()
+	
+// Show an in-app message 
+IterableAPI.inAppManager.show(message: message)
+	
+// Show an in-app message without consuming, i.e., not removing it from the queue
+IterableAPI.inAppManager.show(message: message, consume: false)
 	
-	InApp opens and button clicks are automatically tracked when the notification is called via `spawnInAppNotification`. Using `spawnInAppNotification`, the notification is consumed and removed from the user's in-app messages queue. If you want to retain the messages on the queue, look at using `getInAppMessages` directly. If you use `getInAppMessages` you will need to manage the in-app opens manually in the callback handler.		
-4. ##### Tracking Custom Events
-	Custom events can be tracked using `IterableAPI.track(event:...)` calls.
+```	
+	
+**Objective-C**
+	
+```objc
+// Get the in-app messages list
+NSArray *messages = [IterableAPI.inAppManager getMessages];
+	
+// Show an in-app message 
+[IterableAPI.inAppManager showMessage:message];
+	
+// Show an in-app message without consuming, i.e., not removing it from the queue
+[IterableAPI.inAppManager showMessage:message consume:NO callbackBlock:nil];
+	
+```	
+
+###### When user clicks a button/link in the in-app message
+If the in-app message's clicked button/link's `href` contains a URL (which is usually the case), tapping that button/link will call the `handle` method of `IterableConfig.urlDelegate` (an object of type `IterableURLDelegate`), if one is set. By default if this delegate is not set, tapping the button/link will open Safari with the `href` of the clicked button/link.
+	
+If the in-app message's `href` property contains custom action name (instead of a URL), tapping the  message will call the `handle` method of `IterableConfig.customActionDelegate` (an object that conforms to the `IterableCustomActionDelegate` protocol). If `customActionDelegate` is not set, nothing will happen by default.
```

Comment:
Custom action is specified by passing `itbl://customActionName` as the link URL, we need to clarify that.